### DISCUSSION TITLE
Learn Rate, Loss, TensorBoard Discussion

### DISCUSSION THREAD

**Discussion by JustAnOkapi** [TheLastBen/fast-stable-diffusion#1773] [General]
- Constant `lr_scheduler` is slightly better
- Using TensorBoard

Using constant give slightly lower loss and is easier to resume off. Same conclution as [this reddit post](https://www.reddit.com/r/StableDiffusion/comments/yd56cy/dreambooth_i_compared_all_learning_rate/).

You can enable tensorboard pretty easily with two blocks:
```py
!pip3 uninstall tensorboard # these three fix ValueError: Duplicate plugins for name projector
!pip3 uninstall tensorflow
!pip3 install --ignore-installed tf-nightly # theres probably a correct version to not have to do nightly
```
```py
# !kill 10459
%reload_ext tensorboard
%tensorboard --logdir /content/models/$Session_Name/logs/

import IPython
display(IPython.display.HTML('''
<button id='open_tb'>Open TensorBoard in a new tab.</button> 
<button id='hide_tb'>Hide TensorBoard from colab.</button> 
<script>document.querySelector('#open_tb').onclick = () => { window.open(document.querySelector('iframe').src, "__blank") }
        document.querySelector('#hide_tb').onclick = () => { document.querySelector('iframe').style.display = "none" }</script>'''))
```
This will stay running in the background so you can start training.

I manually reorganize the log files into this structure:
![image](https://user-images.githubusercontent.com/39418801/225226364-cd228c64-f6dd-4666-acd9-bc525ff8da43.png)
(run) (unet or text) (steps)

**Comment by JustAnOkapi** [comment]
- Loss Graphs

Text encoder:
![image](https://user-images.githubusercontent.com/39418801/225226674-c96c83bf-6718-4c92-92fc-04438d4cb8bf.png)

Unet:
![image](https://user-images.githubusercontent.com/39418801/225227015-3fe8c9cb-3818-45cf-95ce-961b3dd8700c.png)

**Comment by JustAnOkapi** [comment]
- Why is loss all over the place, and not going down?
- ~~How come the losses are almost identical when resuming?~~ 
Looks like the seed locked from cross-model testing. After changing the loss no longer overlaps:
![image](https://user-images.githubusercontent.com/39418801/225228600-ca84b28b-f6a3-42b0-a1ad-e44f44e2c2a9.png)

**Comment by TheLastBen** [comment]
in the case of dreambooth, a constant learning rate for a small dataset is not ideal

**Reply by JustAnOkapi** [reply]
seems that the constant option, doesnt even work on unet, it still runs linears anyways. weird

I cant get small datasets to work, it goes straight from not looking like the face, to overfitting.

**Comment by JustAnOkapi** [comment]
why is loss so all over the place anyways, even the avg doesnt go down

**Reply by TheLastBen** [reply]
the loss is displayed for each instance image, not the avg

**Reply by JustAnOkapi** [reply]
yea but smoothed it should be the avg around each image? and that should get better?

**Reply by JustAnOkapi** [reply]
lora training has the classic loss curve
